### ISSUE TITLE
Plugin destroys code completion and other plugins provided typescript types

### ISSUE THREAD

**Issue by Husky-70** [nuxt-modules/i18n#2490]
### Environment

- Operating System: Darwin
- Node Version:     v16.16.0
- Nuxt Version:     3.7.4
- CLI Version:      3.9.0
- Nitro Version:    2.6.3
- Package Manager:  pnpm@8.4.0
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://github.com/Husky-70/types-1 (Without i18n)
https://github.com/Husky-70/types-2 (With i18n)

### Describe the bug

I have created a nuxt project and added dayjs with defineNuxtPlugin inside plugins/dayjs.ts. When I try to use code complition on PhpStorm, $dayjs is missing as soon as I add i18n to the project. This also happens to other plugins.

When i18n is added to the project, code completion only offers $config and $i18n, if I type **$**:
![Bildschirmfoto 2023-10-12 um 13 50 14](https://github.com/nuxt-modules/i18n/assets/50108478/5627fbda-23e4-4c4e-afb6-55b0334c3979)

Without i18n it offers more:
![Bildschirmfoto 2023-10-12 um 14 19 39](https://github.com/nuxt-modules/i18n/assets/50108478/66c740a8-f6a9-46ce-b229-e12314c2d06e)

Types are also gone when i18n is in the project.

According to [this issue](https://github.com/nuxt-modules/i18n/issues/1759), the problem has been solved in beta 8. But it still appears in 8.0.0-rc.5, for me, at least.

Would be thankful for any help on this

**Comment by BobbieGoede** [comment]
I'm unable to replicate this issue with the provided reproductions, it seems to work fine in those. The dependencies in the reproductions differ from your project based on the environment output, maybe updating `nuxt` would do the trick? Not enough info to debug your issue I'm afraid!

Also, in https://github.com/Husky-70/types-1 the `dayjs.ts` plugin is empty, but I copied the plugin from the `types-2` repo to try and replicate the issue.

**Comment by Husky-70** [comment]
> I'm unable to replicate this issue with the provided reproductions, it seems to work fine in those. The dependencies in the reproductions differ from your project based on the environment output, maybe updating `nuxt` would do the trick? Not enough info to debug your issue I'm afraid!
> 
> Also, in https://github.com/Husky-70/types-1 the `dayjs.ts` plugin is empty, but I copied the plugin from the `types-2` repo to try and replicate the issue.

Sorry, I have updated the repro and added correct environment

**Comment by jensk** [comment]
I have the same issue here, with latest nuxt and the recommended vscode volar extension running in takeover mode. 

As soon as '@nuxtjs/i18n' is added to the nuxt config, vscode typescript and vue support stops working.

**Comment by metkm** [comment]
I'm having this issue as well. On a clear project adding `@nuxtjs/i18n` breaks typescript.

**Comment by ChrisGV04** [comment]
Same issue on VS Code. If I add `@nuxtjs/i18n` to the `modules` inside my `nuxt.config.ts`, everything stopped working with Intellisense. And if I only remove this module, everything starts working again.

I believe some peer dependency is causing this because it happened as soon as I updated to Nuxt v3.8.0 and ran `pnpx nuxi upgrade --force`.

I stop getting Intellisense for components inside of the `<template>` tag for `.vue` files and auto-imports aren't found by Typescript

![Screenshot 2023-11-03 at 10 07 54 p m](https://github.com/nuxt-modules/i18n/assets/47041342/bb262641-255e-4f2e-acf7-dd4f9f89a259)

(The dependencies with the yellow line on the left are the ones that changed once I ran `pnpx nuxi upgrade --force`. And yes, I tried `v8.0.0-rc.5` and doesn't work either)

```diff
- '@nuxt/kit': 3.7.4
+ '@nuxt/kit': 3.8.0

- '@vue/compiler-sfc': 3.3.4
+ '@vue/compiler-sfc': 3.3.7

- defu: 6.1.2
+ defu: 6.1.3

- vue-i18n: 9.3.0-beta.27(vue@3.3.4)
+ vue-i18n: 9.3.0-beta.27(vue@3.3.7)

- vue-i18n-routing: 0.13.4(vue-i18n@9.3.0-beta.27)(vue@3.3.4)
+ vue-i18n-routing: 0.13.4(vue-i18n@9.3.0-beta.27)(vue@3.3.7)
```

**Comment by BobbieGoede** [comment]
@Husky-70 @jensk @metkm @ChrisGV04 
Are you still experiencing this issue with the latest edge release installed?

@jensk @metkm @ChrisGV04 
You may be experiencing a similar issue with a different underlying cause, please provide a minimal reproduction so I can take a closer look!

**Comment by metkm** [comment]
> @Husky-70 @jensk @metkm @ChrisGV04 Are you still experiencing this issue with the latest edge release installed?
> 
> @jensk @metkm @ChrisGV04 You may be experiencing a similar issue with a different underlying cause, please provide a minimal reproduction so I can take a closer look!

currently having this issue when running npm install

```ts
Could not resolve module "@intlify/vue-router-bridge/lib/index.mjs"
```

https://github.com/metkm/test
Created a clean project to test it, but I get this issue I posted.

**Comment by Husky-70** [comment]
> @Husky-70 @jensk @metkm @ChrisGV04 Are you still experiencing this issue with the latest edge release installed?
> 
> @jensk @metkm @ChrisGV04 You may be experiencing a similar issue with a different underlying cause, please provide a minimal reproduction so I can take a closer look!

Thank you for your work.
I have added `"@nuxtjs/i18n": "npm:@nuxtjs/i18n-edge@8.0.0-rc.5-28331496.3c75cf9"` to `package.json` on a new project and ran `pnpm install`.

The issue seems to be resolved, as you can see in my screenshot, code completion ans ts works great now.
<img width="677" alt="Bildschirmfoto 2023-11-15 um 17 26 14" src="https://github.com/nuxt-modules/i18n/assets/50108478/07f9aad3-00a9-40be-bdf7-a406d9862be1">


For some reason I wasn't able to get it to work on an existing project. Maybe other plugins forced another version of nuxt/i18n, but have to check this again tomorrow.


Edit: I just tried npm and get the same error as @metkm. `yarn` & `pnpm` are working fine

**Comment by ChrisGV04** [comment]
Hi @BobbieGoede!
I just installed the edge release on my existing project and auto import suggestions are working again! Thank you very much.

**Comment by BobbieGoede** [comment]
@Husky-70 
>For some reason I wasn't able to get it to work on an existing project. Maybe other plugins forced another version of nuxt/i18n, but have to check this again tomorrow.

Removing the lockfile and installing may make things work as well.

@metkm 
It's odd that this only happens when using `npm`, in the meantime the workaround would be to add `@intlify/vue-router-bridge`, `@intlify/vue-i18n-bridge` and `vue-i18n-routing` as dev dependencies. 

Glad to hear that the type inference issue is resolved, so I'm closing this issue. The dependency error for `npm` is a separate issue, so will probably track this separately.

**Comment by metkm** [comment]
> @Husky-70 
> >For some reason I wasn't able to get it to work on an existing project. Maybe other plugins forced another version of nuxt/i18n, but have to check this again tomorrow.
> 
> Removing the lockfile and installing may make things work as well.
> 
> @metkm 
> It's odd that this only happens when using `npm`, in the meantime the workaround would be to add `@intlify/vue-router-bridge`, `@intlify/vue-i18n-bridge` and `vue-i18n-routing` as dev dependencies. 
> 
> Glad to hear that the type inference issue is resolved, so I'm closing this issue. The dependency error for `npm` is a separate issue, so will probably track this separately.
> 

Should I open a new issue?

**Comment by ChrisGV04** [comment]
Yeah @Husky-70! I installed the edge release, deleted the `node_modules` folder and the `pnpm-lock.yaml` file, then installed everything again and it now works fine as before.

```diff
- "@nuxtjs/i18n": "8.0.0-rc.4",
+ "@nuxtjs/i18n": "npm:@nuxtjs/i18n-edge",
```

**Comment by BobbieGoede** [comment]
@metkm

Let's track it here #2556!

**Comment by Husky-70** [comment]
> ```diff
> "@nuxtjs/i18n": "npm:@nuxtjs/i18n-edge",
> ```

Tried to do so multiple times now with different package managers without success. For whatever reason, I can't get it work on an existing project. But I will just wait for the next release of the package.

**Edit:** Got it to work now, it was an old version of the IDE I was using which caused problems. Updating it to the latest version fixed the issue on the existing project now. As I tested it on two different devices, I only got the error on the device with the old IDE version, where I tested it on the existing project.
